src/main/java/com/abrus/user/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.abrus.user.repository;
+
+import com.abrus.user.dto.UserDto;
+import com.abrus.user.model.User;
+import com.abrus.user.util.JpaUtil;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.Query;
+import java.util.List;
+import java.util.Optional;
+
+public class UserRepositoryImpl implements UserRepository {
+    private final EntityManagerFactory emf = JpaUtil.getEntityManagerFactory();
+
+    public User create(User user) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = emf.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return user;
+    }
+
+    public User update(User user) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = emf.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.merge(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return user;
+    }
+
+    @Override
+    public List<User> findAll() {
+        try (EntityManager entityManager = emf.createEntityManager()) {
+            return entityManager
+                    .createQuery("select u from User u", User.class)
+                    .getResultList();
+        }
+    }
+
+    @Override
+    public Optional<User> findById(Long id) {
+        try (EntityManager entityManager = emf.createEntityManager()) {
+            return Optional.ofNullable(entityManager.find(User.class, id));
+        }
+    }
+
+    @Override
+    public List<UserDto> findAllWithRoles() {
```

Comment:
You also can return a List<User>, but return List<UserDto> could also be a valid option